pkg/backend/httpstate/state.go

```diff
@@ -159,18 +159,27 @@ func (u *cloudUpdate) recordEvent(
 	// the user has a rich diff-log they can see when the look at their logs in the service.
 	opts.Color = colors.Raw
 	msg := display.RenderDiffEvent(action, event, seen, opts)
-	if msg == "" {
-		return nil
-	}
 
-	token, err := u.tokenSource.GetToken()
-	if err != nil {
-		return err
-	}
+	for msg != "" {
```

Comment:
Given that `for msg != ""` isn't super obvious what is going on, a quick comment that we are going to write `msg` to the update's log, breaking into smaller chunks as needed, would be super helpful.